www/js/init.js

```diff
@@ -33,8 +33,157 @@
+// A parameter to determine the Settings Store API in use
+params['storeType'] = getBestAvailableStorageAPI();
+params['hideActiveContentWarning'] = getSetting('hideActiveContentWarning') === 'true';
+params['showUIAnimations'] = getSetting('showUIAnimations') ? getSetting('showUIAnimations') === 'true' : true;
+// Maximum number of article titles to return (range is 5 - 50, default 25)
+params['maxSearchResultsSize'] = getSetting('maxSearchResultsSize') || 25;
+// Turns caching of assets on or off and deletes the cache (it defaults to true unless explicitly turned off in UI)
+params['assetsCache'] = getSetting('assetsCache') !== 'false';
+// Turns caching of the PWA's code on or off and deletes the cache (it defaults to true unless the bypass option is set in Expert Settings)
+params['appCache'] = getSetting('appCache') !== 'false';
+// A parameter to set the app theme and, if necessary, the CSS theme for article content (defaults to 'light')
+params['appTheme'] = getSetting('appTheme') || 'light'; // Currently implemented: light|dark|dark_invert|dark_mwInvert|auto|auto_invert|auto_mwInvert|
+// A global parameter to turn on/off the use of Keyboard HOME Key to focus search bar
+params['useHomeKeyToFocusSearchBar'] = getSetting('useHomeKeyToFocusSearchBar') === 'true';
+// A global parameter to turn on/off opening external links in new tab (for ServiceWorker mode)
+params['openExternalLinksInNewTabs'] = getSetting('openExternalLinksInNewTabs') ? getSetting('openExternalLinksInNewTabs') === 'true' : true;
+// A parameter to disable drag-and-drop
+params['disableDragAndDrop'] = getSetting('disableDragAndDrop') === 'true';
+// A parameter to access the URL of any extension that this app was launched from
+params['referrerExtensionURL'] = getSetting('referrerExtensionURL');
+// A parameter to keep track of the fact that the user has been informed of the switch to SW mode by default
+params['defaultModeChangeAlertDisplayed'] = getSetting('defaultModeChangeAlertDisplayed');
+// A parameter to set the content injection mode ('jquery' or 'serviceworker') used by this app
+params['contentInjectionMode'] = getSetting('contentInjectionMode') ||
+    // Defaults to serviceworker mode when the API is available
+    (('serviceWorker' in navigator) ? 'serviceworker' : 'jquery');
+// A parameter to circumvent anti-fingerprinting technology in browsers that do not support WebP natively by substituting images
+// directly with the canvas elements produced by the WebP polyfill [kiwix-js #835]. NB This is only currently used in jQuery mode.
+params['useCanvasElementsForWebpTranscoding'] = null; // Value is determined in uiUtil.determineCanvasElementsWorkaround(), called when setting the content injection mode
 // The key prefix used by the settingsStore.js (see comment there for explanation), but we also need it below
-params['keyPrefix'] = 'kiwixjs-'
+params['keyPrefix'] = 'kiwixjs-';
```

Comment:
You need to move this line (and comment) to BEFORE the first ***use*** of `getSetting()`. It's not enough to have it before the function. It has to be before the function is first called by the code above.